sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -867,6 +872,7 @@ public void deleteDataset(String projectId, String datasetId)
         }
         try {
           sleeper.sleep(nextBackoffMillis);
+          throttlingMilliSeconds.inc(nextBackoffMillis);
```

Comment:
This is for failures. Probably you need to increment the counter for backoff1 for rate limit errors above.

cc: @ihji

Reply:
The retried failures here are transient failures, which I believe include throttling. I have thought about incrementing backoff1 but that is executed in a future (a parallel thread). If we accumulate counters over all threads then I think we will over calculate the number. So I add the counter here in the main thread.

Reply:
Throttling results in rate limit errors, right ? If so that would be captured by backoff1 I think. Prob. Heejong can confirm.

Reply:
My major concern on accumulating `backoff1` is that we may over calculate, because we will be adding time being throttled on all threads.

Reply:
Yes, we should use `backoff1`. Rate limit errors only reach to this point after 2 minutes of backoffs by `backoff1` silently inside of the future. Why do you think it's over-calculated? Each thread is doing its own insert job and it doesn't look strange to me to calculate the total throttling time by adding all backoff times from parallel threads.

Reply:
> Yes, we should use backoff1. Rate limit errors only reach to this point after 2 minutes of backoffs by backoff1 silently inside of the future.

I see. That make sense to me.

> Why do you think it's over-calculated?

Because I am not sure how this metrics is used downstream. I vaguely remember Dataflow autoscaling will use this number divided by the total time spent on work item to yield a fraction which signals throttling. If the total time is not accumulating time spent on all threads then we may over-calculate.

Reply:
Yeah, I think that's a valid concern. We probably need to figure out the time requests are throttled without including backoff due to other errors. Is there a way to get throttled time from all parallel threads and just use the maximum ?

Reply:
> Is there a way to get throttled time from all parallel threads and just use the maximum?

Yes I think this is the right thing to do. Made the change already. PTAL. WDYT, @ihji?